lbz/websocket/request.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import annotations
+
+from multidict import CIMultiDict
+
+from lbz._request import _Request
+from lbz.authentication import User
+from lbz.misc import get_logger
+from lbz.websocket.enums import ActionType
+
+logger = get_logger(__name__)
+
+
+class WebSocketRequest(_Request):
+    """Represents request from Web Socket Secure API Gateway.
+
+    rel: https://docs.aws.amazon.com/apigateway/latest/developerguide/
+    apigateway-websocket-api-mapping-template-reference.html
+    """
+
+    def __init__(
+        self,
+        body: str | bytes | dict,
+        request_details: dict,
+        context: dict,
+        is_base64_encoded: bool,
+        user: User | None = None,
+        headers: CIMultiDict | None = None,
```

Comment:
It looks like more attributes can be moved to the base `Request` class as they are the same in both classes 🤔 

I mean `context`, `user`, `headers`.

Reply:
headers are optional in WSS (only used on connect) User same case, context is a different case
it might become more clear later but in general Message Event is quite different from Connect/Discconet ones.

Reply:
After some testing I moved the User and context but for headers its beneficial to have them on the current level as they will be always present in REST and Optional in WSS and having them always optional messes with typing.